### ISSUE TITLE
What is the reason to have OutputWriter, when there is already OutputInterface?

### ISSUE THREAD

**Issue by greeny** [doctrine/migrations#379]
I don't know any reason why duplicate classes for output. Please correct me if I am wrong. If there is a good reason, I am ok with it.

**Comment by chrisguitarguy** [comment]
`OutputInterface` is from Symfony's console component. It's okay to use that interface in parts of migrations where the code is directly related to the Console component -- like in the [console portion of the application](https://github.com/doctrine/migrations/tree/503ea5cc9c8ae38790a6899d842aaa92d55258bb/lib/Doctrine/DBAL/Migrations/Tools/Console).

Anywhere that doesn't touch the console shouldn't depend on on an interface from the unrelated external package. So stuff like `Configuration`, `Migration, `Version`, and others use an object internal to migrations itself to produce output.

To put it another way, the fact that migrations uses the console component as an user interface is an implementation detail irrelevant to most of the application. `OutputWriter` ensures that implementation detail doesn't leak into the core parts of the library.

**Comment by mikeSimonson** [comment]
Thanks @chrisguitarguy for this very clear explanation.
I hope it's clear enough @greeny ?

**Comment by greeny** [comment]
@chrisguitarguy Thanks for fast answer :) I run into some problems because of this, so I just wanted to know. Its clear now for me.
